### PR TITLE
[QL] Add additional POST params to `/setup_billing_agreement` API call

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

- Add 4 top-level POST params to `v1/paypal_hermes/setup_billing_agreement` if a merchant opts into the PP App Switch flow

### Checklist

- ~Added a changelog entry~

### Authors
@scannillo

### CODE REVIEW

File: UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift [hunk 1]
```diff
@@ -63,4 +67,16 @@ class BTPayPalVaultRequest_Tests: XCTestCase {
         XCTAssertEqual(shippingParams["country_code"], "US")
         XCTAssertEqual(shippingParams["recipient_name"], "Recipient")
     }
+    
+    func testParameters_withEnablePayPalAppSwitchTrue_returnsAllParams() {
+        BTAppContextSwitcher.sharedInstance.universalLink = "some-url"
```
Comment (scannillo) [inline UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift:72]: @jaxdesmarais Should we make[ this property](https://github.com/braintree/braintree_ios/blob/f09dfb24be6cc2236ccb1de844dad89786376563/Sources/BraintreeCore/BTAppContextSwitcher.swift#L20) a `URL` type instead?

Comment (scannillo) [reply UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift:72]: (Outside the scope of this PR, but wanted to ask)

Comment (jaxdesmarais) [reply UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift:72]: 🤔 I am down to change it to a URL type instead.

Comment (scannillo) [reply UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift:72]: Cool, we can do that in another PR. Might as well be more defensive versus just using a `String` type

Comment (jaxdesmarais) [reply UnitTests/BraintreePayPalTests/BTPayPalVaultRequest_Tests.swift:72]: Created DTBTSDK-3659 for this

**Review by jaxdesmarais — APPROVED** [review]

**Comment by scannillo** [comment]
Will wait until https://github.com/braintree/braintree_ios/pull/1227 is merged

**Review by richherrera — APPROVED** [review]

**Pull request merged**